src/github.com/matrix-org/go-neb/services/jira/jira.go

```diff
@@ -0,0 +1,168 @@
+package services
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/andygrunwald/go-jira"
+	"github.com/matrix-org/go-neb/database"
+	"github.com/matrix-org/go-neb/matrix"
+	"github.com/matrix-org/go-neb/plugin"
+	"github.com/matrix-org/go-neb/realms/jira"
+	"github.com/matrix-org/go-neb/types"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+// Matches alphas then a -, then a number. E.g "FOO-123"
+var issueKeyRegex = regexp.MustCompile("([A-z]+)-([0-9]+)")
+var projectKeyRegex = regexp.MustCompile("^[A-z]+$")
+
+type jiraService struct {
+	id     string
+	UserID string
+	Rooms  []string
+}
+
+func (s *jiraService) ServiceUserID() string          { return s.UserID }
+func (s *jiraService) ServiceID() string              { return s.id }
+func (s *jiraService) ServiceType() string            { return "jira" }
+func (s *jiraService) RoomIDs() []string              { return s.Rooms }
+func (s *jiraService) Register() error                { return nil }
+func (s *jiraService) PostRegister(old types.Service) {}
+func (s *jiraService) Plugin(roomID string) plugin.Plugin {
+	return plugin.Plugin{
+		Commands: []plugin.Command{
+			plugin.Command{
+				Path: []string{"jira", "create"},
+				Command: func(roomID, userID string, args []string) (interface{}, error) {
+					// E.g jira create PROJ "Issue title" "Issue desc"
+					if len(args) <= 1 {
+						return nil, errors.New("Missing project key (e.g 'ABC') and/or title")
+					}
+
+					if !projectKeyRegex.MatchString(args[0]) {
+						return nil, errors.New("Project key must only contain A-Z.")
+					}
+
+					pkey := strings.ToUpper(args[0]) // REST API complains if they are not ALL CAPS
+
+					title := args[1]
+					desc := ""
+					if len(args) == 3 {
+						desc = args[2]
+					} else if len(args) > 3 { // > 3 args is probably a title without quote marks
+						joinedTitle := strings.Join(args[1:], " ")
+						title = joinedTitle
+					}
+
+					r, err := s.projectToRealm(userID, pkey)
+					if err != nil {
+						log.WithError(err).Print("Failed to map project key to realm")
+						return nil, errors.New("Failed to map project key to a JIRA endpoint.")
+					}
+					if r == nil {
+						return nil, errors.New("No known project exists with that project key.")
+					}
+
+					iss := jira.Issue{
+						Fields: &jira.IssueFields{
+							Summary:     title,
+							Description: desc,
+							Project: jira.Project{
+								Key: pkey,
+							},
+							// FIXME: This may vary depending on the JIRA install!
+							Type: jira.IssueType{
+								Name: "Bug",
+							},
+						},
+					}
+					cli, err := r.JIRAClient(userID, false)
+					i, res, err := cli.Issue.Create(&iss)
+					if err != nil {
+						log.WithError(err).Print("Failed to create issue")
```

Comment:
Might want to include the project, auth realm, and user in the logging to aid debugging here?